src/components/forms/profile/user-profile-form.tsx

```diff
@@ -15,14 +17,41 @@ import { fetchJson } from "@src/utils/fetch-json";
 
 import "react-datepicker/dist/react-datepicker.css";
 
+type ProfilePayload = Omit<Profile, "id" | "sub" | "updated_at">;
+
+async function updateOrCreateProfile(
```

Comment:
I extracted the function, as I did for addresses.
I also remove the embedded `userProfilePayload` key in the body of the request.

Reply:
I don't really get why you want to extract this into its own function?

Reply:
To avoid duplication and if at any time we want to modify something, we just have to do it here and not in two separate functions

Reply:
But when would you have duplication in this case?

Reply:
it's pretty much the same code for update and create and it makes all the related code in one place.
For example, it avoids having to update the cache for the create or the update, it is done in either cases